### ISSUE TITLE
Font rendering issue

### ISSUE THREAD

**Issue by diveice** [leaningtech/cheerpj-meta#82]
Hello, cheerpJ is an amazing program, it amazingly ran my swing app out of the box. However I have a small problem namely that it doesn't render all unicode arabic characters properly. Specifically some diacritics that aren't common in modern arabic text. When I run my Swing app with Java regularly they're displayed correctly. I don't use a custom view either just the regular JList. Same problem occurs in other components too such as JText. https://imgur.com/utCsHjR.png representation of the text "ٱلرَّحْمَٰنِ ٱلرَّحِيمِ" It seems to have trouble with not just diacritics but also characters like ٱ.

Nevertheless the program is in an excellent state. If this edge case isn't too hard to fix I'd be very grateful! I'm using default fonts by the way, Calibri, Arial, I've tested multiple standard fonts that support arabic text but all suffer from the same problem.

**Comment by alexp-sssup** [comment]
CheerpJ currently have limited support for fonts, in particular CheerpJ cannot access the actual fonts on the system like native Java would. We apologize for this inconvenience, we are a small company and we need to carefully prioritize the features we implement.

We know that our customers have been able to use custom fonts, shipped by them, to render any script correctly. If you have the possibility of modifying the application you could try to use custom fonts in Java and see if that helps.

**Comment by diveice** [comment]
Hi, I've modified the program to load calibri from resources (as a test), using the registerFont method in GraphicsEnvironment after loading the ttf file from resources, however it doesn't seem to work with cheerpJ. This is the only error I see in the console https://imgur.com/NiyGUYg.png

**Comment by alexp-sssup** [comment]
Can you share with us a minimal test case. The crash happens in the low level text layout routines, I suspect there is a code path not currently implemented, but we need a test case to debug and fix the problem.

**Comment by diveice** [comment]
Hi, I've set up a minimal test case, however I noticed now cheerpJ no longer dumps a stack trace but instead stops at a debugging break point without stack trace. I can't compare the stack trace to the old application (the old application now also gets stuck on a break point, but hey both get stuck so I assume the minimal test case reproduces the same problem.


[test-case.zip](https://github.com/leaningtech/cheerpj-meta/files/4476468/test-case.zip)

**Comment by alexp-sssup** [comment]
I cannot reproduce any problem with this test case. Do I need to do something special?

![image](https://user-images.githubusercontent.com/191061/79256371-32229280-7e88-11ea-94c6-8c21f7051454.png)

**Comment by diveice** [comment]
Excuse me, the error I had experienced in the test case I sent to you (breakpoint) I incorrectly assumed it was the same previous error. I think I've hit another bug related to your proguard feature, as I had the proguard flag and systemclipboard flag enabled. When I removed the flags I got the same result as you and the program worked fine. The other error is still there and it gets triggered when I try to write something in arabic. I've modified the test case to write text in arabic instead.  That should reproduce the exact error as in the image I posted in my second comment. 

[test-case2.zip](https://github.com/leaningtech/cheerpj-meta/files/4478068/test-case2.zip)
